### PR TITLE
Handle `__del__` of connection object more gracefully #390

### LINKED ISSUE #390: problem with Exasol driver's WebSocket connection cleanup process
# Summary
When invoking the sql-alchemy engine, it returns the correct result (42) but throws an error during the the WebSocket Connection cleanup process. 

## Reproducing the Issue
Reproducibility:  always

### Steps to reproduce the behavior:
step 1: create file test_sql_alchemy_websocket.py 
```python
# test_sql_alchemy_websocket.py 
from sqlalchemy import create_engine
url = "exa+websocket://user:pw@host:8563/meta?SSLCertificate=SSL_VERIFY_NONE"
engine = create_engine(url)
with engine.connect() as con:
    result = con.execute("select 42").fetchall()
    print(f"{result}")
```

step 2: execute the script:
`python ./resources/scripts/test_sql_alchemy_websocket.py` 

## Expected Behaviour
output should be:
```python
[(42,)]
```
without any further errors

## Actual Behaviour
```python
[(42,)]
Exception ignored in: <function Connection.__del__ at 0x7fe1304df670>
Traceback (most recent call last):
  File "/home/xxxxx/projects/airflow_sbk/.venv/lib/python3.9/site-packages/exasol/driver/websocket/_connection.py", line 151, in __del__
  File "/home/xxxxx/projects/airflow_sbk/.venv/lib/python3.9/site-packages/exasol/driver/websocket/_connection.py", line 127, in close
exasol.driver.websocket._errors.Error: 
```

## Root Cause (optional)
Note that the 42 was returned but there seems a problem with Exasol driver's WebSocket connection cleanup process. Without more descriptive error messages, it's difficult to diagnose the exact issue. This is why I am reaching out to the community for support. Any ideas?

## Context
I am simply following the tutorial on sqlalchemy-exasol and wanted to give the new websocket dialect a try because the odbc dialect is a burden. In particular, many open source tooling require a working sqlalchemy connection.

### System
CentOS Linux release 7.9.2009 (Core)

```
python --version
Python 3.9.18
```

```
pip list 
Package           Version
----------------- -------
cffi              1.16.0
cryptography      41.0.5
greenlet          3.0.1
packaging         23.2
pip               23.0.1
pyasn1            0.5.0
pycparser         2.21
pyexasol          0.25.2
pyodbc            4.0.39
pyOpenSSL         23.3.0
rsa               4.9
setuptools        58.1.0
SQLAlchemy        1.4.44
sqlalchemy-exasol 4.6.0
websocket-client  1.6.4
```

### PR DESCRIPTION
**Pull request by Nicoretti**
Closes #390

### CODE REVIEW

File: test/integration/regression/test_regression_bug390.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+from inspect import cleandoc
+
+pytest_plugins = "pytester"
+
+
+def test_connection_with_block_cleans_up_properly(pytester, exasol_config):
+    config = exasol_config
+    # Because the error only occurs on process shutdown we need to run a test within a test
+    # (We require the result (stderr) of a terminated process triggering the failure.
```
Comment (kaklakariada) [inline test/integration/regression/test_regression_bug390.py:9]: Is it necessary to run a test within a test? Would it be also possible to just run a standalone python script in another process and check its stderr?

File: test/integration/regression/test_regression_bug390.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            query = "SELECT 42;"
+            with engine.connect() as con:
+                result = con.execute(query).fetchall()
+        """
+        ),
+        # fmt: on
+    )
+    r = pytester.runpytest_subprocess()
+    expected = ""
+    actual = f"{r.stderr}"
```
Comment (kaklakariada) [inline test/integration/regression/test_regression_bug390.py]:
```suggestion
    actual = str(r.stderr)
```
❓

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Please add a changelog entry

File: test/integration/regression/test_regression_bug390.py (hunk 1 above)
Comment (Nicoretti) [reply test/integration/regression/test_regression_bug390.py:9]: As discussed

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**